pinot-query-planner/src/test/resources/queries/PinotHintablePlans.json

```diff
@@ -1,33 +1,19 @@
 {
   "pinot_hint_option_tests": {
     "queries": [
-      {
```

Comment:
why not add some planner tests for the tableOptions for partition key + num partitions?

Reply:
see https://github.com/apache/pinot/pull/11234#discussion_r1284542179